### DISCUSSION TITLE
Using gRpc with hono

### DISCUSSION THREAD

**Discussion by shishamo** [honojs/hono#3154] [Q&A] [answered]
Hello guys.

I would like to make a soketio server for web clients an attach a gRpc api that will be use by other micro services.

Is there a way to implement sucharge gRpc to hono?

Thanks!

**Comment by Meess** [comment, marked as answer]
Having an initial glance at the gRPC documentation, atm it doesn't seem to have official support for Typescript / Javascript (see https://grpc.io/docs/#official-support).

As Hono is a JS/TS framework and you want to go RPC you could use the lightweight RPC client of hono (https://hono.dev/docs/guides/rpc) or go with tRPC (see https://github.com/honojs/middleware/tree/main/packages/trpc-server).

**Reply by shishamo** [reply]
Thanks @Meess for your reply~
